helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataReader.java

```diff
@@ -134,6 +147,7 @@ public synchronized void handleChildChange(String s, List<String> list) {
 
     // Subscribe data changes again because some children might have been deleted or added
     _zkClient.unsubscribeAll();
+    _zkClient.subscribeChildChanges(MetadataStoreRoutingConstants.ROUTING_DATA_PATH, this);
```

Comment:
After making the fix, `handleChildChange` and `handleDataDeleted` are completely equal now. I added a comment on line 127 based on what you told me to justify the duplication - sometimes the callbacks may go missing. Should we consider calling `handleDataDeleted` from `handleChildChange`, for example, to reduce duplication? @narendly

Reply:
Resolved offline.